src/services/inlayHints.ts

```diff
@@ -740,11 +735,75 @@ export function provideInlayHints(context: InlayHintsContext): InlayHint[] {
                         visitForDisplayParts(propertySignature.type);
                     }
                     break;
+                case SyntaxKind.MethodSignature:
+                    const methodSignature = node as MethodSignature;

```

Comment:
```suggestion
                    Debug.assertNode(node, isMethodSignature);

```
would be a bit cleaner, imo. Though I get we're not using the assertion tech elsewhere in this switch yet (and, optimally, ofc, we'd just have a union type instead of `Node`), so it's not really critical.